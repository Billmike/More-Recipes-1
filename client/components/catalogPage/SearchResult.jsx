import React from 'react';
import PropTypes from 'prop-types';

import RecipeCatalogCard from '../common/RecipeCatalogCard';
import Spinner from '../common/Spinner';

const propTypes = {
  recipes: PropTypes.shape({
    isFetchingRecipes: PropTypes.bool,
    searchResult: PropTypes.arrayOf(PropTypes.shape()),
  }).isRequired,
};

/**
 * Function representing search result component
 *
 * @param {object} props - The properties passed to the component
 *
 * @returns {object} React element
 */
const SearchResult = (props) => {
  const { recipes: { isFetchingRecipes, searchResult } } = props;

  return (
    <div>
      {isFetchingRecipes
        ? <div className="center-align">
          <Spinner size="big" />
        </div>
        : <div>
          {searchResult && searchResult.map((recipe, index) =>
            (<RecipeCatalogCard
              {...props}
              key={recipe.id}
              index={index}
              recipe={recipe}
            />))}
        </div>}
    </div>
  );
};

SearchResult.propTypes = propTypes;

export default SearchResult;
