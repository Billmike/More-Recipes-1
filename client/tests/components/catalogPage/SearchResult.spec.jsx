import React from 'react';

import SearchResult from '../../../components/catalogPage/SearchResult';

describe('SearchResult component', () => {
  it('should render component', () => {
    const props = {
      recipes: {
        isFetchingRecipes: false,
        searchResult: [],
      },
    };
    const wrapper = shallow(<SearchResult {...props} />);

    expect(wrapper).toBeDefined();
    expect(wrapper.find('RecipeCatalogCard').exists()).toEqual(false);
    expect(wrapper.find('Spinner').exists()).toEqual(false);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render component with spinner when searching for recipe', () => {
    const props = {
      recipes: {
        isFetchingRecipes: true,
        searchResult: [],
      },
    };
    const wrapper = shallow(<SearchResult {...props} />);

    expect(wrapper).toBeDefined();
    expect(wrapper.find('RecipeCatalogCard').exists()).toEqual(false);
    expect(wrapper.find('Spinner').exists()).toEqual(true);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render component with search results', () => {
    const props = {
      recipes: {
        isFetchingRecipes: false,
        searchResult: [{
          title: 'Beans',
          category: 'Main',
          description: 'Tasty beans',
          preparationTime: 85,
          ingredients: 'Beans, onions, palm oil, salt',
          directions: 'Do this. Do that',
          views: 350,
          upvotes: 48,
          downVotes: 2,
          favorites: 58,
        }],
      },
    };
    const wrapper = shallow(<SearchResult {...props} />);

    expect(wrapper).toBeDefined();
    expect(wrapper.find('RecipeCatalogCard').exists()).toEqual(true);
    expect(wrapper.find('Spinner').exists()).toEqual(false);
    expect(wrapper).toMatchSnapshot();
  });
});
