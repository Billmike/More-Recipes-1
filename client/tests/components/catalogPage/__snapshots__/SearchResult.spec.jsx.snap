// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResult component should render component 1`] = `
<div>
  <div />
</div>
`;

exports[`SearchResult component should render component with search results 1`] = `
<div>
  <div>
    <RecipeCatalogCard
      index={0}
      recipe={
        Object {
          "category": "Main",
          "description": "Tasty beans",
          "directions": "Do this. Do that",
          "downVotes": 2,
          "favorites": 58,
          "ingredients": "Beans, onions, palm oil, salt",
          "preparationTime": 85,
          "title": "Beans",
          "upvotes": 48,
          "views": 350,
        }
      }
      recipes={
        Object {
          "isFetchingRecipes": false,
          "searchResult": Array [
            Object {
              "category": "Main",
              "description": "Tasty beans",
              "directions": "Do this. Do that",
              "downVotes": 2,
              "favorites": 58,
              "ingredients": "Beans, onions, palm oil, salt",
              "preparationTime": 85,
              "title": "Beans",
              "upvotes": 48,
              "views": 350,
            },
          ],
        }
      }
    />
  </div>
</div>
`;

exports[`SearchResult component should render component with spinner when searching for recipe 1`] = `
<div>
  <div
    className="center-align"
  >
    <Spinner
      size="big"
    />
  </div>
</div>
`;
